src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementMenuEntityUpdater.java

```diff
@@ -130,43 +131,36 @@ public void onOrganizationIdUpdateRequest(@Observes OrganizationIdUpdateRequest
   public void timeout(Timer timer) {
     if (!stopped) {
       if (!queue.isEmpty()) {
-        updateManagementMenus(queue.remove(0));
+        MenuIdUpdateRequest updateRequest = queue.remove(0);
+        DefaultApi api = managementApi.getApi(updateRequest.getOrganizationId());
+        updateManagementMenu(api, updateRequest.getOrganizationId(), updateRequest.getId());
       }
 
       startTimer(SystemUtils.inTestMode() ? 1000 : TIMER_INTERVAL);
     }
   }
 
-  private void updateManagementMenus(OrganizationId organizationId) {
-    List<fi.otavanopisto.mwp.client.model.Menu> managementMenus = listManagementMenus(organizationId);
-    for (fi.otavanopisto.mwp.client.model.Menu managementMenu : managementMenus) {
-      Menu menu = updateManagementMenu(organizationId, managementMenu);
-      MenuId menuId = new MenuId(organizationId, KuntaApiConsts.IDENTIFIER_NAME, menu.getId());
-      
-      List<Menuitem> managementMenuItems = listManagementMenuItems(organizationId, managementMenu);
-      for (Menuitem managementMenuItem : managementMenuItems) {
-        updateManagementMenuItem(organizationId, menuId, managementMenuItem);
-      }
+  private void updateManagementMenu(DefaultApi api, OrganizationId organizationId, MenuId managementMenuId) {
+    ApiResponse<fi.otavanopisto.mwp.client.model.Menu> response = api.kuntaApiMenusIdGet(managementMenuId.getId());
+    if (response.isOk()) {
+      updateManagementMenu(api, organizationId, response.getResponse());
+    } else {
+      logger.warning(String.format("Finding organization %s menu failed on [%d] %s", managementMenuId.getId(), response.getStatus(), response.getMessage()));
     }
   }
 
-  private List<fi.otavanopisto.mwp.client.model.Menu> listManagementMenus(OrganizationId organizationId) {
-    DefaultApi api = managementApi.getApi(organizationId);
+  private void updateManagementMenu(DefaultApi api, OrganizationId organizationId, fi.otavanopisto.mwp.client.model.Menu managementMenu) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "updateManagementMenu" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)